packages/plugin-ext/src/plugin/languages/rename.ts

```diff
@@ -18,9 +18,8 @@ import URI from 'vscode-uri/lib/umd';
 import * as theia from '@theia/plugin';
 import * as Converter from '../type-converters';
 import * as model from '../../api/model';
-import { DocumentsExtImpl } from '@theia/plugin-ext/src/plugin/documents';
```

Comment:
that's a good catch, how did you find it?

Reply:
tslint was complaining that `@theia/plugin-ext` was implicitly depending on itself so I 
checked as to how/why :see_no_evil:

Reply:
Was it a warning as well, it would be good to have it as error.

Reply:
It was a warning, I guess we need a custom rule to find out if an extension is trying to import istelf this way.